cmd/ci-images-mirror/main.go

```diff
@@ -320,6 +338,69 @@ func main() {
 	logrus.Info("Process ended gracefully")
 }
 
+func validateConfig(config *quayiociimagesdistributor.CIImagesMirrorConfig, registryConfig string, ciOperatorconfigPath string) error {
+	r, err := regexp.Compile("docker.io")
+	if err != nil {
+		return fmt.Errorf("failed to compile regex: %w", err)
+	}
+	for nsWithName, source := range config.SupplementalCIImages {
+		if source.Image != "" {
+			if match := r.MatchString(source.Image); match {
+				return fmt.Errorf("source image %s is from docker.io", source.Image)
+			}
+			if !isAccessible(source.Image, registryConfig) {
+				return fmt.Errorf("source image %s is not accessible", source.Image)
+			}
+		}
+		if isPromoted(nsWithName, ciOperatorconfigPath) {
+			return fmt.Errorf("target %s would overwrite a promoted tag", nsWithName)
+		}
+		if isDuplicated(nsWithName, config) {
+			return fmt.Errorf("target %s is duplicated", nsWithName)
+		}
+	}
+	return nil
+}
+
+func isPromoted(sourceISTagRef string, ciOperatorconfigPath string) bool {
+	abs, err := filepath.Abs(ciOperatorconfigPath)
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to determine absolute CI Operator configuration path")
+	}
+	err = config.OperateOnCIOperatorConfigDir(abs, func(cfg *api.ReleaseBuildConfiguration, metadata *config.Info) error {
+		for _, isTagRef := range release.PromotedTags(cfg) {
+			if sourceISTagRef == isTagRef.ISTagName() {
+				return fmt.Errorf("source image %s is promoted", sourceISTagRef)
+			}
+		}
+		return nil
+	})
+	return err != nil
+}
+
+func isDuplicated(nsWithName string, config *quayiociimagesdistributor.CIImagesMirrorConfig) bool {
```

Comment:
I believe you could not even do this today?
IIRC, we have a yaml lint to avoid this.
Could you please make a PR to the release repo and see if any failures show up?

Reply:
"A new tag is added to override an existing tag."
By this I mean:
We check if the target exists in QCI for a new tag.
So you need to know somehow the list of new tags coming with a PR.

To achieve that, we may parse the config file from the PR and the config file in the current release repo and see the difference.